pandas/tests/frame/test_alter_axes.py

```diff
@@ -1056,6 +1056,38 @@ def test_reindex_signature(self):
                               "limit", "copy", "level", "method",
                               "fill_value", "tolerance"}
 
+    def test_droplevel(self):
+        # GH20342
+        df = pd.DataFrame([
+            [1, 2, 3, 4],
+            [5, 6, 7, 8],
+            [9, 10, 11, 12]
+        ])
+        df = df.set_index([0, 1]).rename_axis(['a', 'b'])
+        ser = df.iloc[:, 0]
+        df.columns = pd.MultiIndex.from_tuples([('c', 'e'), ('d', 'f')],
+                                               names=['level_1', 'level_2'])
+
+        # test that dropping of a level in DataFrame index works
+        expected_df_no_level_a_in_index = df.reset_index('a', drop=True)
+        actual_df_no_level_a_in_index = df.droplevel('a')
+        assert_frame_equal(expected_df_no_level_a_in_index,
+                           actual_df_no_level_a_in_index)
+
+        # test that dropping of a level in DataFrame columns works
```

Comment:
can you parameterize these on axis and also try with axis='columns',  and axis='index'